provider/virtual_network/network_base.py

```
@@ -149,20 +149,29 @@ def _ping():
+    :param tap_mtu: tap mtu setting value
     """
     # Create tap device with ip command
     tap_cmd = f'ip tuntap add mode tap user {user} group {user} name ' \
               f'{tap_name} {flag};ip link set {tap_name} up;' \
               f'ip link set {tap_name} master {bridge_name}'
     # Execute command as root
-    process.run(tap_cmd, shell=True, verbose=True)
+    if session:
+        session.cmd(tap_cmd, timeout=60)
+    else:
+        process.run(tap_cmd, shell=True, verbose=True)
```

Ruff (0.14.1)

Function call with `shell=True` parameter identified, security issue

(S604)